### ISSUE TITLE
mkdocs-merge affected by setuptools moving to 70.2.0

### ISSUE THREAD

**Issue by mzarucki** [ovasquez/mkdocs-merge#20]
Hi @ovasquez,

Suddenly `mkdocs-merge` stopped working for us:

```
Cloning into 'ps-instructions'...
$ mkdocs-merge run . ps-instructions
Attempting to merge site: ps-instructions
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs-merge", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/mkdocsmerge/__main__.py", line 44, in run
    merge.run_merge(master_site, sites, unify_sites, print_func=click.echo)
  File "/usr/local/lib/python3.9/site-packages/mkdocsmerge/merge.py", line 35, in run_merge
    new_navs = merge_sites(sites, master_docs_root, unify_sites, print_func)
  File "/usr/local/lib/python3.9/site-packages/mkdocsmerge/merge.py", line 105, in merge_sites
    dir_util.copy_tree(old_site_docs, new_site_docs, update=1)
  File "/usr/local/lib/python3.9/site-packages/setuptools/_distutils/dir_util.py", line 146, in copy_tree
    mkpath(dst, verbose=verbose)
  File "/usr/local/lib/python3.9/site-packages/setuptools/_distutils/dir_util.py", line 82, in mkpath
    _path_created.add(abs_head)
AttributeError: 'dict' object has no attribute 'add'
```
https://gitlab.cern.ch/cms-tsg/fog/documentation/-/jobs/40713402

Could you please help us understand the underlying issue?

Thanks a lot.

Cheers,
Mateusz

**Comment by mzarucki** [comment]
One notable difference: `setuptools`  moved from `70.1.1` to `70.2.0` wrt. to the last successful pipeline

**Comment by ovasquez** [comment]
@mzarucki thanks for catching this! I'll investigate if only the update is necessary and publish a new version.

**Comment by ovasquez** [comment]
@mzarucki I have published version [0.9.0](https://github.com/ovasquez/mkdocs-merge/releases/tag/0.9.0) which should fix the reported issue. I was able to reproduce the bug locally and with 0.9.0 the problem no longer happens.

**Comment by bodacious-bill** [comment]
I'm getting a similar bug with setuptools==80.1.0

`  File "[proj]/.venv/lib/python3.13/site-packages/mkdocsmerge/merge.py", line 54, in merge_sites
    dir_util._path_created.clear()
    ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'setuptools._distutils.dir_util' has no attribute '_path_created'`

Downgrading to 70.1.1 fixed it. Also, when managing packages with uv, setuptools did not install as a dependency. I had to add it manually.

**Comment by ovasquez** [comment]
@bodacious-bill I haven't used uv so I don't know if there's anything different there. Does the problem also happen with pip?

**Comment by ovasquez** [comment]
@bodacious-bill thanks for reporting the issue. I have added a permanent fix for this issue on version `0.10.1`.
